src/Compilers/CSharp/Test/Semantic/Semantics/InterpolationTests.cs

```diff
@@ -5227,5 +5227,950 @@ .locals init (CustomHandler1 V_0)
+            AssertEx.Equal("System.Runtime.CompilerServices.InterpolatedStringHandlerArgumentAttribute",
+                               cParam.GetAttributes().Single().AttributeClass.ToTestDisplayString());
+            Assert.Empty(cParam.InterpolatedStringHandlerArgumentIndexes);
+        }
+
+        [Fact]
+        public void InterpolatedStringHandlerArgumentAttributeError_MismatchedType()
+        {
+            var code = @"
+using System.Runtime.CompilerServices;
+class C
+{
+    static void M(int i, [InterpolatedStringHandlerArgumentAttribute(""i"")] CustomHandler c) {}
+}
+
+public partial struct CustomHandler
+{
+    public CustomHandler(int literalLength, int formattedCount, string s) : this() {}
+}
+";
+
+            var handler = GetCustomHandlerType("CustomHandler", "partial struct", useBoolReturns: true);
+
+            var comp = CreateCompilation(new[] { code, InterpolatedStringHandlerArgumentAttribute, handler }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (5,27): error CS9011: Could not find a constructor on 'CustomHandler' that matches the parameters specified in the InterpolatedStringHandlerArgumentAttribute.
+                //     static void M(int i, [InterpolatedStringHandlerArgumentAttribute("i")] CustomHandler c) {}
+                Diagnostic(ErrorCode.ERR_CouldNotFindApplicableInterpolatedStringHandlerConstructor, @"InterpolatedStringHandlerArgumentAttribute(""i"")").WithArguments("CustomHandler").WithLocation(5, 27)
+            );
+
+            var cParam = comp.SourceModule.GlobalNamespace.GetTypeMember("C").GetMethod("M").Parameters.Skip(1).Single();
+            AssertEx.Equal("System.Runtime.CompilerServices.InterpolatedStringHandlerArgumentAttribute",
+                           cParam.GetAttributes().Single().AttributeClass.ToTestDisplayString());
+            Assert.Empty(cParam.InterpolatedStringHandlerArgumentIndexes);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(", out bool success")]
+        public void InterpolatedStringHandlerArgumentAttribute_SingleArg(string extraConstructorArg)
+        {
+            var code = @"
+using System.Runtime.CompilerServices;
+public class C
+{
+    public static void M(int i, [InterpolatedStringHandlerArgumentAttribute(""i"")] CustomHandler c) {}
+}
+
+public partial struct CustomHandler
+{
+    public CustomHandler(int literalLength, int formattedCount, int i" + extraConstructorArg + @") : this()
+    {
+" + (extraConstructorArg != "" ? "success = true;" : "") + @"
+    }
+}
+";
+
+            var handler = GetCustomHandlerType("CustomHandler", "partial struct", useBoolReturns: true);
+
+            var comp = CreateCompilation(new[] { code, InterpolatedStringHandlerArgumentAttribute, handler }, parseOptions: TestOptions.RegularPreview);
+            CompileAndVerify(comp, sourceSymbolValidator: verifier, symbolValidator: verifier);
```

Comment:
>CompileAndVerify [](http://example.com/codeflow?start=12&length=16)

Consider VerifyDiagnostics on result of CompileAndVerify calls. This helps catch unexpected warnings, etc.